pkg/apis/certificates/types.go

```diff
@@ -83,3 +87,58 @@ type CertificateSigningRequestList struct {
 
 	Items []CertificateSigningRequest `json:"items,omitempty"`
 }
+
+type CertificateSigningProfileList struct {
+	unversioned.TypeMeta `json:",inline"`
+	unversioned.ListMeta `json:"metadata,omitempty"`
+
+	Items []CertificateSigningProfile `json:"itmes,omitempty"`
+}
+
+// CertificateSigningProfile specifies certian certificate attributes
+// that a certificate signing requestor can request of a certificate.
+// The Certificate Authority will take this profile into account when
+// issuing the requested certificate.
+type CertificateSigningProfile struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	// expiry denotes a duration the certificate should be issued for
+	// after which it becomes invalid.
+	Expiry unversioned.Duration
+
+	// allowedUsages specifies a set of usage context the key will be
+	// valid for.
+	// See: https://tools.ietf.org/html/rfc5280#section-4.2.1.3
+	//      https://tools.ietf.org/html/rfc5280#section-4.2.1.12
+	AllowedUsages []KeyUsage
+}
+
+type KeyUsage string
+
+const (
+	UsageSigning            = "signing"
```

Comment:
are there standard formats for these? are there constants we can use, either from golang or from cfssl, or elsewhere?